src/parcsr_ls/par_relax.c

```diff
@@ -1119,7 +1118,7 @@ hypre_BoomerAMGRelax7Jacobi( hypre_ParCSRMatrix *A,
    /*-----------------------------------------------------------------
     * Perform Matvec Vtemp = w * (f - Au)
     *-----------------------------------------------------------------*/
-   if (zero_u)
+   if (hypre_ParVectorAllZero(u))
```

Comment:
How about moving this check inside `hypre_ParCSRMatrixMatvec`? Then, other parts of the code code potentially benefit from the fact that a vector is all zeros.

Reply:
let's talk about this in the meeting later. Thanks!